src/cpp/benders/benders_core/BendersOptions.cpp

```diff
@@ -70,7 +46,7 @@ void BendersOptions::read(std::string const &file_name)
 #undef BENDERS_OPTIONS_MACRO
 		}
 
-		if (SLAVE_WEIGHT != "UNIFORM" && SLAVE_WEIGHT != "CONSTANT")
+		if (SLAVE_WEIGHT != SLAVE_WEIGHT_UNIFORM && SLAVE_WEIGHT != SLAVE_WEIGHT_CONSTANT)
 		{
```

Comment:
Use enum classes

Reply:
in fact, it would be nicer to use enum class. The problem is that SLAVE_WEIGHT is a member of BendersOptions class, **declared**, and **initialized** by macros using **BendersOptions.hxx** file.